spyder/api/plugins.py

```diff
@@ -8,69 +8,165 @@
 spyder.api.plugins
 ==================
 
-Here, 'plugins' are Qt widgets that can make changes to Spyder's
-main window and call other plugins directly.
+Here, 'plugins' are Qt widgets that can make changes to Spyder's main window
+and call other plugins directly.
+
+There are two types of plugins available.
+* SpyderPlugin is a plugin that does not create a new dock/pane on Spyder
+  main window
+* SpyderPluginWidget is a plugin that does create a new dock/pane.
 """
 
+# Standard library imports
+import inspect
+import os
+import sys
+
 # Third party imports
-from qtpy.QtCore import Signal, Slot
-from qtpy.QtWidgets import QWidget
+from qtpy.QtCore import QObject, Qt, Signal, Slot
+from qtpy.QtGui import QCursor, QIcon, QKeySequence
+from qtpy.QtWidgets import (QAction, QApplication, QDockWidget, QMainWindow,
+                            QMenu, QMessageBox, QShortcut, QToolButton,
+                            QWidget)
+import qdarkstyle
 
 # Local imports
+from spyder.config.base import _
+from spyder.config.gui import get_color_scheme, get_font, is_dark_interface
+from spyder.config.manager import CONF
 from spyder.config.user import NoDefault
-from spyder.plugins.base import BasePluginMixin, BasePluginWidgetMixin
+from spyder.plugins.base import PluginWindow
+from spyder.py3compat import configparser, is_text_string
 from spyder.utils import icon_manager as ima
+from spyder.utils.qthelpers import (MENU_SEPARATOR, add_actions, create_action,
+                                    create_toolbutton, set_menu_icons,
+                                    toggle_actions)
+from spyder.widgets.dock import SpyderDockWidget
 
 
-# =============================================================================
-# SpyderPlugin
-# =============================================================================
-class BasePlugin(BasePluginMixin):
-    """
-    Basic functionality for Spyder plugins.
+class PluginBaseMixin:
```

Comment:
PluginBaseMixin sounds odd. It should be BasePluginMixin

Reply:
Agreed!